math_unit.pas

```diff
@@ -25303,11 +25318,18 @@ procedure drawtimbcl(retcurve: boolean);       // mark timber centre-line
     // shift required to rotation point, to suit the curving..
     y_curmod := y_curtimb - yeq;
 
-    if midline = False then
-      calc_fill_timber_mark(EMarkCode(4 + shove_this))  // 4 = centre-line,  44 = selected timber.
-    else
-      calc_fill_timber_mark(EMarkCode(14 + shove_this));
-    // 14,54 = timber centre-line, use solid line if drawing rail centrelines. (for rivets?)
+    if midline then
+       case timberStatus of
+         eTS_Normal:
+           code := eMC_14_TimberCLMidline;
+         eTS_Shoved:
+           code := eMC_54_ShovingTimberCLMidline;
+         eTS_Selected:
+           code := eMC_54_ShovingTimberCLMidline;
+       end;
```

Comment:
What about the midline *false* condition?

Reply:
Oh crumbs. This code should be the same as that a little frther down in drawtimber().  I am ssure it was at one point. Corrected version coming up ...